scripts/case_change_service.py

```diff
@@ -0,0 +1,65 @@
+# coding: utf-8
+#
+# Copyright 2020 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Service for changing the case of string or dict to camelCase"""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import re
+
+
+def camelize_string(string):
+    """Changes the case of string from snake_case to camelCase.
+
+    Args:
+        string: str. The string whose case is to changed to camelCase
+        from snake_case.
+
+    Returns:
+        The string after changing the case of input string to camelCase.
+    """
+
+    snake_case_re = re.compile(r'([^\-_\s])[\-_\s]+([^\-_\s])')
+
+    return ''.join([
+        string[0].lower() if not string[:2].isupper() else string[0],
+        snake_case_re.sub(
+            lambda m: m.group(1) + m.group(2).upper(), string[1:]),
+    ])
+
+
+def camelize(obj):
+    """Changes the case of the keys of dict from snake_case to camelCase.
+
+    Args:
+        obj: dict or list of dicts. If the object is a dict it changes
```

Comment:
I'd recommend splitting this function, you're doing too much here. One function to process list of dicts and the other to process just dicts.

Reply:
Actually the object could be anything. So, we don't know if the object has a list of strings or dicts. That is it is possible that a dict contains list of dicts.

Reply:
What if someone puts in a set?

Reply:
It would return the same set.

Reply:
well, that kind of makes the function meaningless then right? What I'm trying to get here is that your function is trying to do too many things. It's better to say that the function does this one thing well which is to camelize things in a list or camelize keys in a dictionary.

Reply:
The real purpose of the function is to camelize the keys of a dict. The list part is introduced if a dict something this this appears.
```json
{
    key_one: [
    {
        key_two: 0
    }
    ]
}
```

So now the functions checks recursively for the dicts.

Reply:
In that case, I'd recommend asserting that you can accept only lists or dicts.

Reply:
No, because the values in the dict are also passed in the function if, they're not dicts or lists they need to be returned as they are.

Reply:
Okay, make sure you write tests to cover all possible cases.

Reply:
Looking at this, it doesn't make sense to say that the object you're passing in is a dict or list of dics, since your function takes in more than that. So just change the type to object.

Reply:
Done